File: A/modules/Fly.py
import math
class Fly:
    def __init__(self, x, y, z, v=0, direction=0):
        '''
        x,y,z:初始位置
        v:速度
        direction:飞行方向（弧度制，0为x轴正方向，逆时针旋转）
         '''
        self.x=x
        self.y=y
        self.z=z
        self.vx=v*math.cos(direction)
        self.vy=v*math.sin(direction)
        self.vz=0
    def fly_tick(self,dt):#平飞一段时间
        self.x+=self.vx*dt
        self.y+=self.vy*dt
    def drop_tick(self,dt): #自由落体一段时间
        self.x+=self.vx*dt
        self.y+=self.vy*dt
        self.z=self.z-0.5*9.81*dt*dt+self.vz*dt
        self.vz-=9.81*dt
    def fly(self,t):#平飞未来t秒后的位置
        F_new=Fly(self.x,self.y,self.z,0,0)
        F_new.vx=self.vx
        F_new.vy=self.vy
        F_new.vz=self.vz
        F_new.x+=F_new.vx*t
        F_new.y+=F_new.vy*t
        return F_new
    def drop(self,t):#自由落体未来t秒后的位置
        F_new=Fly(self.x,self.y,self.z,0,0)
        F_new.vx=self.vx
        F_new.vy=self.vy
        F_new.vz=self.vz
        F_new.x+=F_new.vx*t
        F_new.y+=F_new.vy*t
        F_new.z=F_new.z-0.5*9.81*t*t+F_new.vz*t
        F_new.vz-=9.81*t
        return F_new

File: A/rust/src/modules/fly.rs
// Rust version of Fly.py
pub struct Fly {
    pub vx: f64,
    pub vy: f64,
    pub vz: f64,
    pub x: f64,
    pub y: f64,
    pub z: f64,
}
const FY: [[f64; 3]; 6] = [
    [0.0, 0.0, 0.0],
    [17800.0, 0.0, 1800.0],
    [12000.0, 1400.0, 1400.0],
    [6000.0, -3000.0, 700.0],
    [11000.0, 2000.0, 1800.0],
    [13000.0, -2000.0, 1300.0],
];

impl Fly {
    pub fn new(direction: f64, v: f64, x: f64, y: f64, z: f64) -> Self {
        Fly { vx: v * direction.cos(), vy: v * direction.sin(), vz: 0.0, x, y, z }
    }
    pub fn new_id(id:usize,direction: f64, v: f64) -> Self {

        let idx = if id < FY.len() { id } else { 0 };
        let pos = FY[idx];
        Fly {
            vx: v * direction.cos(),
            vy: v * direction.sin(),
            vz: 0.0,
            x: pos[0],
            y: pos[1],
            z: pos[2],
        }
    }


    pub fn fly_tick(&mut self, t: f64) {
        self.x += self.vx * t;
        self.y += self.vy * t;
    }
    pub fn fly(&mut self,t:f64)->Fly {
        Fly {
            vx: self.vx,
            vy: self.vy,
            vz: self.vz,
            x: self.x + self.vx * t,
            y: self.y + self.vy * t,
            z: self.z,
        }
    }
    pub fn drop_tick(&mut self, t: f64) {
        self.x += self.vx * t;
        self.y += self.vy * t;
        self.z = self.z + self.vz * t - 0.5 * 9.81 * t * t;
        self.vz -= 9.81 * t;
    }
    pub fn drop(&mut self,t:f64)->Fly {
        let new_z = self.z + self.vz * t - 0.5 * 9.81 * t * t;
        let new_vz = self.vz - 9.81 * t;
        Fly {
            vx: self.vx,
            vy: self.vy,
            vz: new_vz,
            x: self.x + self.vx * t,
            y: self.y + self.vy * t,
            z: new_z,
        }
    }

}
